### PR TITLE
Refactor _update_many

### PR DESCRIPTION
**Pull request by kouk**
goals are:

1. avoid duplicating the field splitting in `insert_or_update_many`
2. support arbitrary iterable arguments for `keys`, `update_fields` and
`exclude_fields` (e.g. tuples no longer work after #6)
4. abstract the model meta stuff in separate methods
3. make everything a bit more clear

### CODE REVIEW

File: djangobulk/bulk.py [hunk 1]
```diff
@@ -10,18 +10,41 @@
...
+
+    :param model: A model class.
+    :param field_names: An iterable of field names that should be returned if
+        they match field in the model. If empty, no fields are returned.
+        If None the model's primary key is returned.
+    :returns: A list of fields that should be used in a WHERE
+    """
+    # If no fields are specified, use the primary key by default
+    if field_names is None:
+        return [model._meta.pk.name]
```
Comment (igavriil) [inline djangobulk/bulk.py]: `model_keys` should be instances of `django.db.models.fields`, so 
```
if field_names is None:
    return [model._meta.pk]
```

**Review by igavriil — CHANGES_REQUESTED** [review]

File: djangobulk/bulk.py [hunk 2]
```diff
@@ -10,18 +10,41 @@
...
+    field_names = set(field_names)
+    fields = []
+    for f in model._meta.fields:
+        should_skip_field = field_names and f.name in field_names
```
Comment (igavriil) [inline djangobulk/bulk.py]: `field_names` contains the field names that should be returned. Here you are excluding them.
`should_keep_field = field_names and f.name in field_names`

**Review by igavriil — CHANGES_REQUESTED** [review]

File: djangobulk/bulk.py (hunk 2 above)
Comment (kouk) [reply djangobulk/bulk.py]: duh :)

**Review by igavriil — APPROVED** [review]
LGTM

**Pull request merged**